pkg/resources/collector/workloads.go

```diff
@@ -46,13 +46,17 @@ func (r *Reconciler) aggregatorConfigMapName() string {
 	return fmt.Sprintf("%s-aggregator-config", r.collector.Name)
 }
 
-func (r *Reconciler) getDaemonConfig(loggingReceivers []string) otel.NodeConfig {
+func (r *Reconciler) getDaemonConfig(loggingReceivers []string, traceReceivers []string) otel.NodeConfig {
```

Comment:
The trace receivers are push-based, so I don't think they should be configured as daemons, but I don't see how else we could do. Any thoughts?

Reply:
The way I see it there are two options for the daemonset:
 1. Not using the daemonset at all: 
 So something along the lines of : 
```go
func (r *Reconciler) requiresDaemoset() {
  return r.Config.Logging != nil || r.Config.Metricss != nil
}
```

And in the method where the daemonset is reconciled, something along the lines of : 
```
if !r.requiresDaemonset() {
  resources.Absent(daemonset)
} else {
   // ...
   resources.Present()
}
```
2. (better) collecting the internal traces from the open-telemetry collector:

The open telemetry collector collects self-telemetry from the following service:
```yaml
service:
  telemetry:
      logs:
        level: {{ .LogLevel }}
      metrics:
        level: none
      traces:
        // TODO
```

Note : you should definitely include the self-telemetry for at least the aggregator